lib/_emerge/Scheduler.py

```diff
@@ -1953,26 +1953,21 @@ def _schedule_tasks_imp(self):
         @return: True if state changed, False otherwise.
         """
 
-        state_change = 0
-
         while True:
             if not self._keep_scheduling():
-                return bool(state_change)
+                return False
 
             if (
-                self._choose_pkg_return_early
-                or self._merge_wait_scheduled
+                self._merge_wait_scheduled
                 or (self._jobs and self._unsatisfied_system_deps)
                 or not self._can_add_job()
                 or self._job_delay()
```

Comment:
A reason to include self._choose_pkg_return_early in this condition is to skip unnecessary calls to self._can_add_job() and self._job_delay() when self._choose_pkg() will return None anyway. It saves some cpu time spent scheduling.

Reply:
I see, that is a good reason to keep it. I've dropped the according commit.